extensions/pkg/controller/bastion/controller.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	extensionspredicate "github.com/gardener/gardener/extensions/pkg/predicate"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+const (
+	// FinalizerName is the bastion controller finalizer.
+	FinalizerName = "extensions.gardener.cloud/bastion"
+	// ControllerName is the name of the controller
+	ControllerName = "bastion_controller"
+)
+
+// AddArgs are arguments for adding a Bastion controller to a manager.
+type AddArgs struct {
+	// Actuator is a Bastion actuator.
+	Actuator Actuator
+	// ControllerOptions are the controller options used for creating a controller.
+	// The options.Reconciler is always overridden with a reconciler created from the
+	// given actuator.
+	ControllerOptions controller.Options
+	// Predicates are the predicates to use.
+	// If unset, GenerationChangedPredicate will be used.
+	Predicates []predicate.Predicate
+	// Type is the type of the resource considered for reconciliation.
+	Type string
+}
+
+// DefaultPredicates returns the default predicates for a controlplane reconciler.
+func DefaultPredicates(ignoreOperationAnnotation bool) []predicate.Predicate {
+	if ignoreOperationAnnotation {
+		return []predicate.Predicate{
+			predicate.GenerationChangedPredicate{},
+		}
+	}
+
+	return []predicate.Predicate{
+		predicate.Or(
+			extensionspredicate.HasOperationAnnotation(),
+			extensionspredicate.LastOperationNotSuccessful(),
```

Comment:
This predicate is probably not needed because the `Bastion` is not created as part of a shoot reconciliation but completely separate, so the last operation of the `Shoot` should not play a role here, or?

Reply:
Will removing this cause the requeue to not happen anymore? At least the AWS provider's controller relies on requeues to wait for the bastion to come up, for example.

Reply:
Sorry, I made a mistake. I was wrongly thinking that the `LastOperationNotSuccessful` predicate is about the last operation of the related `Shoot` (for whatever reason, I was mixing it up with https://github.com/gardener/gardener/blob/77e09dd522d7bfb682cc35d0be437201b606061b/extensions/pkg/predicate/predicate.go#L82-L85). Please add it again.